tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/core/SentenceTagger.scala

```diff
@@ -0,0 +1,279 @@
+      version <- googleUnigramConfig.get[Int]("version")
+    } yield {
+      new DatastoreGoogleNGram(groupName, artifactName, version, 1000)
+    }
+  }
+  private lazy val verbnet: Option[Verbnet] = {
+    val taggersConfig =
+      ConfigFactory.parseFile(new File(taggersConfigFile))
+    for {
+      verbnetConfig <- taggersConfig.get[Config]("verbnet")
+      groupName <- verbnetConfig.get[String]("group")
+      artifactName <- verbnetConfig.get[String]("name")
+      version <- verbnetConfig.get[Int]("version")
+    } yield {
+      new Verbnet(groupName, artifactName, version)
+    }
+  }
+
+  /** Initializes a sentence tagger from a SentenceTaggerInitializer recipe.
+    *
+    * @param initializer the initialization recipe
+    * @return the initialized tagger
+    */
+  def initialize(initializer: SentenceTaggerInitializer): SentenceTagger = {
+    initializer match {
+      case FactoriePostaggerInitializer(useCoarseTags) =>
+        NLPStackPostagger(factorieTagger, useCoarseTags)
+      case StanfordPostaggerInitializer(useCoarseTags) =>
+        NLPStackPostagger(stanfordTagger, useCoarseTags)
+      case SimplePostaggerInitializer(configFile) =>
+        SimplePostagger.load(configFile)
+      case LexicalPropertiesTaggerInitializer =>
+        IndependentTokenSentenceTagger(LexicalPropertiesTagger)
+      case TokenPositionTaggerInitializer =>
+        TokenPositionTagger
+      case BrownClustersTaggerInitializer(clusters) =>
+        IndependentTokenSentenceTagger(BrownClustersTagger(clusters))
+      case KeywordTaggerInitializer(keywords) =>
+        IndependentTokenSentenceTagger(KeywordTagger(keywords))
+      case GoogleUnigramTaggerInitializer(tagType) =>
+        IndependentTokenSentenceTagger(GoogleUnigramTagger(googleNgrams.get, tagType))
+      case VerbnetTaggerInitializer =>
+        VerbnetTagger(verbnet.get)
+    }
+  }
+
+  /** Applies a sequence of sentence taggers to a sentence, tagging each token with the union
+    * of their tags.
+    *
+    * @param sentence the sentence we want to tag
+    * @param taggers the taggers we want to apply
+    * @return a tagged sentence
+    */
+  def tagWithMultipleTaggers(sentence: Sentence, taggers: Seq[SentenceTagger]): TaggedSentence = {
+    val taggings = taggers map { tagger =>
+      tagger.tag(sentence)
+    }
+    TaggedSentence(
+      sentence,
+      (Range(0, sentence.tokens.size) map { tokIndex =>
+      (tokIndex, taggings.toSet flatMap { tagging: TaggedSentence =>
+        tagging.tags.getOrElse(tokIndex, Set())
+      })
+    }).toMap
+    )
+  }
+
+  /** Tags all sentences from a given sentence source.
+    *
+    * Note that this function is multithreaded.
+    *
+    * @param tagger the tagger to apply
+    * @param sentenceSource the sentence source
+    * @return an iterator over the resulting tagged sentences
+    */
+  def tagSentenceSource(
+    tagger: SentenceTagger,
+    sentenceSource: SentenceSource
+  ): Iterator[TaggedSentence] = {
```

Comment:
This should take an implicit execution context, instead of using a local import.